py/desisim/test/test_quickbrick.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+import numpy as np
+from astropy.io import fits
+from desisim.scripts import quickbrick
+import desispec.io
+
+class TestQuickBrick(unittest.TestCase):
+    
+    #- Pick random test directory
+    def setUp(self):
+        self.testdir = 'test-{uuid}'.format(uuid=uuid1())
+
+    #- Cleanup test files if they exist
+    def tearDown(self):
+        if os.path.exists(self.testdir):
+            rmtree(self.testdir)
+            
+    #- Run basic test of quickbrick and check its outputs
+    @unittest.skipIf('TRAVIS_JOB_ID' in os.environ, 'Skipping memory hungry quickbrick/specsim test on Travis')
+    def test_quickbrick(self):
+        brickname = 'test1'
+        nspec = 3
+        cmd = "quickbrick -b {} --objtype ELG -n {} --outdir {}".format(brickname, nspec, self.testdir)
+        args = quickbrick.parse(cmd.split()[1:])
+        results = quickbrick.main(args)
+        #- Do the bricks exist?
+        for channel in ['z', 'r', 'b']:
+            brickfile = '{}/brick-{}-{}.fits'.format(self.testdir, channel, brickname)
+            self.assertTrue(os.path.exists(brickfile))
+            
+        #- Check one of them for correct dimensionality of contents
+        brickfile = '{}/brick-b-{}.fits'.format(self.testdir, brickname)
+        with fits.open(brickfile) as fx:
+            self.assertEqual(fx['FLUX'].shape[0], nspec)
+            self.assertEqual(fx['FLUX'].shape, fx['IVAR'].shape)
+            self.assertEqual(fx['FLUX'].shape[1], fx['WAVELENGTH'].shape[0])
+            self.assertEqual(fx['FLUX'].shape[0], fx['RESOLUTION'].shape[0])
+            self.assertEqual(fx['FLUX'].shape[1], fx['RESOLUTION'].shape[2])
+            self.assertEqual(len(fx['FIBERMAP'].data), nspec)
+            self.assertEqual(fx['_TRUEFLUX'].shape, fx['FLUX'].shape)
+            self.assertTrue(np.all(fx['_TRUTH'].data['OBJTYPE'] == 'ELG'))
+        
+        #- Make sure that desispec I/O functions can read it
+        frame = desispec.io.read_frame(brickfile)
+        self.assertEqual(frame.flux.shape[0], nspec)
+        brick = desispec.io.Brick(brickfile)
+        self.assertEqual(brick.get_num_spectra(), nspec)
+        brick.close()
+
+    #- Test quickbrick with a bunch of options
+    @unittest.skipIf('TRAVIS_JOB_ID' in os.environ, 'Skipping memory hungry quickbrick/specsim test on Travis')
+    def test_quickbrick_options(self):
+        brickname = 'test2'
+        nspec = 3
+        cmd = "quickbrick -b {} --objtype BGS -n {} --outdir {}".format(brickname, nspec, self.testdir)
+        cmd = cmd + " --airmass 1.5 --verbose --zrange-bgs 0.1 0.2"
+        cmd = cmd + " --moon-phase 0.1 --moon-angle 30 --moon-zenith 20"
+        args = quickbrick.parse(cmd.split()[1:])
+        results = quickbrick.main(args)
+        brickfile = '{}/brick-b-{}.fits'.format(self.testdir, brickname)
+        truth = fits.getdata(brickfile, '_TRUTH')        
+        z = truth['TRUEZ']
+        self.assertTrue(np.all((0.1 <= z) & (z <= 0.2)))
```

Comment:
Could also check that the median ivar is within some tolerance of the expected value, to catch any changes that significantly change the results.